### PR TITLE
Allow excludedatabases when using file_per_database

### LINKED ISSUE #449: Add option to exclude certain databases
For our purposes, we don't need to backup the mysql, performance_schema or information_schema databases, so would like to exclude them from backups.

We can modify the [mysqlbackup.sh.erb](https://github.com/puppetlabs/puppetlabs-mysql/blob/master/templates/mysqlbackup.sh.erb) template with something like:

``` sh
mysql -s -r -N -e 'SHOW DATABASES' <% if ! @excludedatabases.empty? %> | grep -v '^\(<% @excludedatabases.join('|') %>)$' <% end %> | while read dbname
```

### PR DESCRIPTION
**Pull request by HT43-bqxFqB**
Fixes #449 #461

### CODE REVIEW

File: manifests/backup/mysqldump.pp [hunk 1]
```diff
@@ -33,6 +33,7 @@
   $compression_command      = 'bzcat -zc',
   $compression_extension    = '.bz2',
   $backupmethod_package     = undef,
+  Array[Strings] $excludedatabases = [],
```
Comment (HT43-bqxFqB) [inline manifests/backup/mysqldump.pp:36]: We can indent the other parameters as well.

**Comment by HT43-bqxFqB** [comment]
Is anyone able to review this PR?

File: spec/classes/mysql_backup_mysqldump_spec.rb [hunk 2]
```diff
@@ -71,6 +71,21 @@ class { 'mysql::server': }
...
+      context 'with file_per_database and excludedatabases' do
+        let(:params) do
+          {
+            'file_per_database' => true,
+            'excludedatabases'  => [ 'information_schema' ],
+          }.merge(default_params)
+        end
+
+        it {
+          is_expected.to contain_file('mysqlbackup.sh').with_content(
```
[the remark this answers is not in the document]
Comment (LukasAud) [reply spec/classes/mysql_backup_mysqldump_spec.rb:84]: SLES-15 errors are unrelated

[the remark this answers is not in the document]
Comment (HT43-bqxFqB) [reply spec/classes/mysql_backup_mysqldump_spec.rb:84]: I'm on it. The problem seems to be the
```
<% @excludedatabases.join('|') %>
```
part.

[the remark this answers is not in the document]
Comment (HT43-bqxFqB) [reply spec/classes/mysql_backup_mysqldump_spec.rb:84]: Should be fixed by now. A `=` was missing in the template code.

**Comment by HT43-bqxFqB** [comment]
Commits are now squashed :heavy_check_mark:

File: manifests/backup/mysqldump.pp (hunk 1 above)
Comment (LukasAud) [reply manifests/backup/mysqldump.pp]: If you could indent the other parameters, that would be great. Once that is done, I will happy to merge this PR 👍

**Review by LukasAud — APPROVED** [review]

**Comment by LukasAud** [comment]
Hey @HT43-bqxFqB, I'm happy to merge this one. Do not worry about the indentation, we'll make a quick fix as soon as we release it.

Thanks for your contribution! 👍

**Pull request merged**